test/parallel/test-stream-reduce.js

```diff
@@ -0,0 +1,112 @@
+'use strict';
+
+const common = require('../common');
+const {
+  Readable,
+} = require('stream');
+const assert = require('assert');
+
+function sum(p, c) {
+  return p + c;
+}
+
+{
+  // Does the same thing as `(await stream.toArray()).reduce(...)`
+  (async () => {
+    const tests = [
+      [[], sum, 0],
+      [[1], sum, 0],
+      [[1, 2, 3, 4, 5], sum, 0],
+      [[...Array(100).keys()], sum, 0],
```

Comment:
`[...it]` is not needed since Readable.from can take an iterator, no?

Reply:
Yes but as an iterator it is consumed once read - so the comparison reduce the following line gets an already finished iteartor.